packages/back-end/src/jobs/refreshFactTableColumns.ts

```diff
@@ -138,7 +138,38 @@ export async function runRefreshColumnsQuery(
 
   const typeMap = new Map<string, FactTableColumnType>();
   const jsonMap = new Map<string, JSONColumnFields>();
-  determineColumnTypes(result.results).forEach((col) => {
+
+  result.columns?.forEach((col) => {
+    // If the underlying SQL engine returned the datatype, use it
+    if (col.dataType !== undefined) {
+      // For JSON, only return if we have the field information, otherwise skip
+      // so we can infer from the returned data
+      if (
+        col.dataType === "json" &&
+        col.fields !== undefined &&
+        col.fields.length > 0
+      ) {
+        typeMap.set(col.name, "json");
+        jsonMap.set(
+          col.name,
+          col.fields.reduce(
+            (acc, field) => ({
+              ...acc,
+              [field.name]: {
+                datatype: field.dataType,
+              },
+            }),
+            {},
+          ),
+        );
+      } else if (col.dataType !== "json") {
+        typeMap.set(col.name, col.dataType);
+      }
+    }
+  });
+
+  const columnsToSkip = new Set(typeMap.keys());
+  determineColumnTypes(result.results, columnsToSkip).forEach((col) => {
```

Comment:
There is another small regression.  On main, selecting a JSON string would be inferred as type `json`, but here it would just be set to type `string`.

Instead of `columnsToSkip`, maybe the whole existing typeMap should be passed in.  Then, `determineColumnTypes` can decide if it needs to further refine any of the column types based on the sample row values.